js/admin.js

```diff
@@ -8,20 +8,30 @@
 		currentConfig: '1',
 		providerIds: '1',
 
-		_getAppConfig: function (key) {
-			return $.ajax({
-				type: 'GET',
-				url: OC.linkToOCS('apps/provisioning_api/api/v1', 2) + 'config/apps' + '/user_saml/' + key + '?format=json'
-			});
-		},
 		init: function(callback) {
-			this._getAppConfig('providerIds').done(function (data){
-				if (data.ocs.data.data !== '') {
-					OCA.User_SAML.Admin.providerIds = data.ocs.data.data;
-					OCA.User_SAML.Admin.currentConfig = OCA.User_SAML.Admin.providerIds.split(',').sort()[0];
+			var xhr = new XMLHttpRequest();
+			xhr.open('GET', OC.generateUrl('/apps/user_saml/settings/providers'));
+			xhr.setRequestHeader('Content-Type', 'application/json');
+			xhr.setRequestHeader('requesttoken', OC.requestToken);
+
+			xhr.onload = function () {
+				var response = JSON.parse(xhr.response);
+				if (xhr.status >= 200 && xhr.status < 300) {
+					if (response.providerIds !== "") {
+						OCA.User_SAML.Admin.providerIds += ',' + response.providerIds;
+						OCA.User_SAML.Admin.currentConfig = OCA.User_SAML.Admin.providerIds.split(',').sort()[0];
```

Comment:
sorting might not be required, it seems the backend already does it: https://github.com/nextcloud/user_saml/pull/558/files#diff-7466884a39494cea3f50d3aaa2c0e3aed4adefb1630992261d6d56e6b812f8fbR56

(sorting could be slower for a lot of entries, but probably not a big issue)